packages/arb-bridge-peripherals/contracts/tokenbridge/ethereum/gateway/L1CustomGateway.sol

```diff
@@ -113,20 +113,46 @@ contract L1CustomGateway is L1ArbitrumExtendedGateway, ICustomGateway {
         return l1ToL2Token[l1ERC20];
     }
 
+    /**
+     * @notice Allows L1 Token contract to trustlessly register its custom L2 counterpart. (other registerTokenToL2 method allows excess eth recovery from _maxSubmissionCost and is recommended)
+     * @param _l2Address counterpart address of L1 token
+     * @param _maxGas max gas for L2 retryable exrecution
+     * @param _gasPriceBid gas price for L2 retryable ticket
+     * @param  _maxSubmissionCost base submission cost  L2 retryable tick3et
+     * @return Retryable ticket ID
+     */
+    function registerTokenToL2(
+        address _l2Address,
+        uint256 _maxGas,
+        uint256 _gasPriceBid,
+        uint256 _maxSubmissionCost
+    ) external payable virtual returns (uint256) {
+        return
+            this.registerTokenToL2(
```

Comment:
I think we should instead make `registerTokenToL2` public and call it that way.
`this. registerTokenToL2` makes an external evm call